rclcpp/include/rclcpp/experimental/executors/events_executor/priority_events_queue.hpp

```diff
@@ -0,0 +1,173 @@
+// Copyright 2023 iRobot Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__EXPERIMENTAL__EXECUTORS__EVENTS_EXECUTOR__PRIORITY_EVENTS_QUEUE_HPP_
+#define RCLCPP__EXPERIMENTAL__EXECUTORS__EVENTS_EXECUTOR__PRIORITY_EVENTS_QUEUE_HPP_
+
+#include <condition_variable>
+#include <functional>
+#include <mutex>
+#include <queue>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "rclcpp/experimental/executors/events_executor/events_queue.hpp"
+
+namespace rclcpp
+{
+namespace experimental
+{
+namespace executors
+{
+
+struct PriorityEvent
+{
+  int priority;
+  rclcpp::experimental::executors::ExecutorEvent event;
+};
+
+struct ComparePriorities : public std::binary_function<PriorityEvent, PriorityEvent, bool>
+{
+  _GLIBCXX14_CONSTEXPR
```

Comment:
I am pretty sure, this will either not compile on windows, or not give the wanted result.